opm/simulators/flow/DamarisWriter.hpp

```diff
@@ -259,15 +269,81 @@ class DamarisWriter : public EclGenericWriter<GetPropType<TypeTag, Properties::G
+                dam_err_ = DamarisOutput::setPosition(name.c_str(), rank_,
+                                                      this->elements_rank_offsets_[rank_]);
+            }
+
+            // Call damaris_write() for all available variables
+            for ( auto damVar : localCellData ) 
+            {
+               // std::map<std::string, data::CellData>
+              const std::string& name = damVar.first ;
+              
+              std::unordered_set<std::string>::const_iterator is_in_set = wanted_vars_set_.find ( name );
+              
+              if ((is_in_set != wanted_vars_set_.end() ) || (wanted_vars_set_.size() == 0)) {
+                  data::CellData  dataCol = damVar.second ;
+                  OpmLog::debug(fmt::format("Name of Damaris Variable       : ( rank:{})  name: {}  ",  rank_, name));
+                  
+                  // It does not seem I can test for what type of data is present (double or int)
+                  // in the std::variant within the data::CellData, so I will use a try catch block. 
+                  // Although, only MPI_RANK and GLOBAL_CELL_INDEX are set as integer types (in the 
+                  // XML file) so it is a moot point. 
+                  // We could use damaris_get_type() to check what the type is specified as
+                  // within the Damaris XML file
+                  try {
+                    if (dataCol.data<double>().size() >= this->numElements_) {
+                        dam_err_ = DamarisOutput::write(name.c_str(), rank_,
+                                                        dataCol.data<double>().data()) ;
+                    }
+                  }
+                  catch (std::bad_variant_access const& ex) {
+                    // Not a std::vector<double>
+                    if (dataCol.data<int>().size() >= this->numElements_) {
+                        dam_err_ = DamarisOutput::write(name.c_str(), rank_,
+                                                      dataCol.data<int>().data()) ;
+                    }
+                  }
+              }
+              
+
             }
+            
+           /*   
+            Code for when we want to pass to Damaris the single cell 'block' data variables
+            auto mybloc = damarisOutputModule_->getBlockData() ;
+            for ( auto damVar : mybloc ) {
+               // std::map<std::string, data::CellData>
+              const std::string name = std::get<0>(damVar.first) ;
+              const int part = std::get<1>(damVar.first) ;
+              double  dataCol = damVar.second ;
+              std::cout << "Name of Damaris Block Varaiable : (" << rank_ << ")  "  << name  << "  part : " << part << "  Value : "  << dataCol <<  std::endl ;  
+            } 
+            
+            dam_err_ =  DamarisOutput::endIteration(rank_);
+            */
+            if (this->damarisOutputModule_->getPRESSURE_ptr() != nullptr) 
```

Comment:
I do not understand what the logic is here? Also it seems a bit non-generic?

Reply:
Me either. However, if this is not present, then the first iteration (iteration 0 in Damaris) is missed, and the GLOBAL_CELL_INDEX variable is not written. There is probably a better way to do this test, however, this works, and is derived from the original PRESSURE only code.

Reply:
We can leave it like this for now.